test-utils/junit5-extension/src/main/java/org/operaton/bpm/engine/test/junit5/ParameterizedTestExtension.java

```diff
@@ -158,22 +166,42 @@ public Object resolveParameter(ParameterContext parameterContext,
 							return parameters[index];
 						}
 					},
-					// A TestInstanceFactory that uses our parameters to create a new instance.
+					// TestInstanceFactory to create a new test instance using the parameters
 					new TestInstanceFactory() {
 						@Override
 						public Object createTestInstance(TestInstanceFactoryContext factoryContext,
 								ExtensionContext extensionContext) throws TestInstantiationException {
 							try {
 								Class<?> testClass = extensionContext.getRequiredTestClass();
-								// we assume the test class has a single constructor.
+								// assume the test class has a single constructor.
 								Constructor<?> constructor = testClass.getDeclaredConstructors()[0];
 								constructor.setAccessible(true);
-								// Use the parameters provided by our invocation context.
 								return constructor.newInstance(parameters);
 							} catch (Exception e) {
 								throw new TestInstantiationException("Could not create test instance", e);
 							}
 						}
+					},
+					// TestInstancePostProcessor for field injection using @Parameter(X)
+					new TestInstancePostProcessor() {
+						@Override
+						public void postProcessTestInstance(Object testInstance, ExtensionContext context)
+								throws Exception {
+							// Iterate over declared fields and inject values for those annotated with
+							// @Parameter
+							for (Field field : testInstance.getClass().getDeclaredFields()) {
+								Parameter parameterAnnotation = field.getAnnotation(Parameter.class);
+								if (parameterAnnotation != null) {
+									int index = parameterAnnotation.value();
+									if (index < 0 || index >= parameters.length) {
+										throw new ExtensionConfigurationException("Index " + index
+												+ " is out of bounds for the provided parameters array.");
+									}
+									field.setAccessible(true);
+									field.set(testInstance, parameters[index]);
```

Comment:
do you think the field accessibility should be reset after injecting the field? Likely does not make any difference, though, just feels more correct.

Reply:
Hmmm. Don't feel the same way. It would be in the same way more correct to add a AfterEach method to your test classes that nulls the fields all the time when you have an BeforeEach that fills the fields. I would not do add this code as it not really adds value.